internal/v1/handler_post_compose_test.go

```diff
@@ -566,10 +566,60 @@ func TestComposeImageErrorsWhenCannotParseResponse(t *testing.T) {
 		},
 	}
 	respStatusCode, body := tutils.PostResponseBody(t, "http://localhost:8086/api/image-builder/v1/compose", payload)
-	require.Equal(t, 500, respStatusCode)
+	require.Equal(t, http.StatusInternalServerError, respStatusCode)
 	require.Contains(t, body, "Internal Server Error")
 }
 
+// This test case queries the image-builder for a non existing type of the os distribution
+// The mock of osbuild-composer returns correct code and payload but still the error
+// should be intercepted by image-builder
+func TestComposeImageErrorsWhenDistributionNotExists(t *testing.T) {
+	id := uuid.New()
+	apiSrv := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if "Bearer" == r.Header.Get("Authorization") {
+			w.WriteHeader(http.StatusUnauthorized)
+			return
+		}
+		require.Equal(t, "Bearer accesstoken", r.Header.Get("Authorization"))
+		w.Header().Set("Content-Type", "application/json")
+		w.WriteHeader(http.StatusCreated)
+		result := composer.ComposeId{
+			Id: id,
+		}
+		err := json.NewEncoder(w).Encode(result)
+		require.NoError(t, err)
+	}))
+	defer apiSrv.Close()
+
```

Comment:
I think this can be removed, image builder doesn't contact composer in this scenario.

Reply:
I have checked and you are right. Code updated:)